### DISCUSSION TITLE
Freetz-ng baut kein Image

### DISCUSSION THREAD

**Discussion by hookdr** [Freetz-NG/freetz-ng#1063] [Anfänger]
Hallo, ich möchteinImage mit oscam für meine Frritzbox 7590ax bauen. Komme leider nicht weiter.

Firmware wird jetzt gebaut, Oscam lässt sich nicht starten

**Comment by fda77** [comment]
Das liegt an imatig, das ist voller bugs und verursach dazu in freetz fehler

Siehe dazu: https://freetz-ng.github.io/freetz-ng/ADDONS.html#steigerung-der-unzul%C3%A4nglichkeit

Öffne am besten hier ein issue bei @MasterRoCcO

**Comment by SaMMy-lacht** [comment]
Sorry, aber das liegt an deinen Patchen!

Egal welchen Treiber man aus den Kernel benutzen will. Es wird zu ein Abbruch führen. Auch das Erstellen danach ohne einen Treiber aus dem Kernel kann man nur beheben, in dem man make kernel-clean macht.  Und das betrifft alle Geräte bis auf die 6591 und 6660 im FOS8 Bereich.

**Reply by fda77** [reply]
Wie gesagt, lass deinen Quatsch weg und es geht

**Reply by Conan179** [reply]
Schmeis die Patches von fda77 bei deinem Fork raus, fertig.

**Reply by SaMMy-lacht** [reply]
Was redest du da! Es geht halt nicht! Es liegt an deinen Patchen! 
Hier noch mal ein Video! Sauberes freetz-ng 
https://www.mediafire.com/file/ewx0p4waipwgp0c/7590AX_NEW_330_CRAP_FIX_PATCH.mkv/file

ca. ab 18 min wird es dann interessant

**Reply by fda77** [reply]
pfff, lads doch auf youtube

**Reply by SaMMy-lacht** [reply]
Teste es selber! Egal was man anmacht im Kernel, es wird dir ein Abbruch bringen! 

Fix deine Patche und gut ist!

**Reply by fda77** [reply]
ich mag mich wirklich nicht mit deinem script auseinander setzen

**Reply by SaMMy-lacht** [reply]
Junge wie oft den noch. freetz-ng ist der schuldige. Teste es selber und du wirst es sehen oder schau dir das Video an. 
Da alle Video und TESTS mit ein sauberen neu geladen freetz-ng gemacht worden sind, kannst du es nicht auf anderes schieben. Aber soll mir auch egal sein. Es ist ja nicht mein Problem. Da ich es mir ja gefixt habe.

Noch andere Videos
7590AX
https://www.mediafire.com/file/6pm0ug7xm4n58gd/7590AX_mit_patch_fix.mkv/file
7530AX
https://www.mediafire.com/file/j96sc1zngfpgyjh/7530AX_mit_patch_fix.mkv/file
ein sauberes freetz-ng erst ohne fixe nach Abbruch wurde das gemacht make kernel-clean und die fixe rein und schau an, es geht.

**Reply by Conan179** [reply]
Dafür das es "nicht dein Problem" macht du aber viel Wind. Fix es doch einfach in deinem Fork und brüste dich damit "Mein Fork hat einen fehler gefixt wo freetz-ng drin hat!"

**Reply by fda77** [reply]
sag doch einfach wie die Fehlermeldung ist oder mach einen PR mit fix, falls wirklich nötig

**Reply by Conan179** [reply]
@fda77 Glaubst du Wirklich er tut seine Paydarbeit öffentlich zugänglich machen? Da friert eher die Hölle ein, bevor er das macht.

**Reply by SaMMy-lacht** [reply]
Was für Pay Arbeit?
Aber danke für solche Unterstellungen! 

Wenn fda77 es gefixt hat, werde ich mal was dazu sagen!

**Reply by fda77** [reply]
Kann er ja, aber mich damit immer zu nerven

**Reply by fda77** [reply]
Ausserdem weiss ich eh was drin ist :] ccid,pscsc updates usw
Es gibt halt eine grobes Missverständnis (im deb) wie open source geht
Ich schick einfach alle weiter in sein repo

**Reply by Conan179** [reply]
Rofl aus Heiser Luft kommt von dir doch eh nicht.
Wen du jetzt deine sh... nicht ablassen kannst, kannst du es nie.

**Reply by fda77** [reply]
> SaMMy-lacht:
> Wenn fda77 es gefixt hat, werde ich mal was dazu sagen!

Wie gesagt, ich mache dein Spielchen NICHT mit. 
Du kannst es also lassen

**Reply by SaMMy-lacht** [reply]
Dann werden alle FOS8 (die es bis jetzt gibt) bis auf 6591 und 6660 keine Kernel nutzen können!

**Reply by fda77** [reply]
Bedauerlich

**Reply by Conan179** [reply]
Auser natürlich bei deinem Paidfork, da läuft es ja, weil der Fehler dort schon behoben worden ist.

**Reply by SaMMy-lacht** [reply]
Gut das man den Kernel ja nicht nur für das dumme oscam brauch.

**Reply by fda77** [reply]
ich hab kein ahnung was du gemacht hast, aber es ist so falsch. wie es richtig geht muss du selbst herausfinden. ich hab dazu weder zeit noch lust

**Reply by SaMMy-lacht** [reply]
Wie oft den noch, ich habe da nichts gemacht. Weil du denn Fehler gemacht hast. Und das nicht zum ersten mal.

**Reply by Conan179** [reply]
Und nicht das erste mal das DU auf einen auf dicke Hose machst und sagt, nenenene ich weis was, was du nicht, ich bin besser wie ihr alle, weil ich angepisst bin, wegen dem nicht deb support.

Ich könnte jetzt zum xigten mal fragen, warum du nicht hilfst, bringt aber nichts.

**Comment by fda77** [comment]
Ich hab mal die vorhin hinzugefügte 7590 mit ein paar Kernelmodulen gebaut. Wie zu erwarten gab es KEINE Probleme. Dummschwätzer! Also geht mir bitte nicht weiter auf die Nerven. Such dir doch einfach Freunde

![image](https://github.com/user-attachments/assets/1ed0390f-3e61-4ba0-8eb3-b759b59cc4a0)

**Reply by SaMMy-lacht** [reply]
Ich habe Freunde! Ich gebe dir nun das 2 Mal den Tipp weil du es ja sonst nicht glauben wirst. Schaue dir mal wieder die 2 unterschiedlichen Patche namens 310-no_diffconfig.patch. Das haten wir doch schon 11 Jun 2024

**Reply by fda77** [reply]
Wenn du Freunde hast, warum nevst du nicht die???

**Reply by SaMMy-lacht** [reply]
Weil die ja nicht so Mist verzapfen! Schau dir es in der 7590AX an und in der 6591 oder 6660. Und schau an es gibt wieder unterschiedliche wo der eine den Fehler macht und der andere durchlauft. 

Und da habe ich es damals auch freigegeben 
https://www.mediafire.com/file/cc7eivvldabiv3d/pipefail_fehler.txt.crypt/file

gpg --passphrase da_mir_sonst_ja_später_keiner_glaubt_hier_der_Beweis_erszthfnjzftu8rf6thzf5t678ir --no-symkey-cache --cipher-algo AES256 --batch -o pipefail_fehler.txt.crypt -d pipefail_fehler.txt.crypt

und das steht da drin

Schon eine sehr komische Sache, das es von dem 310-no_diffconfig.patch 2 Versionen gibt
der eine der den pipefail Fehler ab den source files 7.8 und 6670 bringt und einen der für die 7690 ist und keinen Fehler bringt. Der Fehler ist aber noch in anderen drin.

Hier erst mal der Inhalt der fehlerhaften Datei (6670)
```
--- linux-5.15/GNUmakefile
+++ linux-5.15/GNUmakefile
@@ -196,17 +196,8 @@
 		@rm -f .kernelvariables .kernelvariables.old)
 	+@$(call __kbuild,$(and $(MAKECMDGOALS),$@))
 
-ifeq ($(is-oot-kmod),)
-	@# Skip config diff for out-of-tree module builds
-    ifeq ($(words $(wildcard $(sort $(AVM_KCONFIG_CONFIG) $(KCONFIG_CONFIG)))),2)
-	$(if $(filter-out $(no-config-diff-targets),$@),\
-		@scripts/diffconfig $(AVM_KCONFIG_CONFIG) $(KCONFIG_CONFIG) | \
-		 sed -re '1{s|^|Linux config has changed compared to $(AVM_KCONFIG_CONFIG):\n|}' >&2)
-    endif
-else
 	$(if $(filter modules_install,$@),\
 		@$(MAKE) --no-print-directory modules_install.post)
-endif
 
 
 ifeq ($(is-oot-kmod),)
```

Nun Der Inhalt der fehlerfreien Datei (7690)
```
--- linux-5.4/GNUmakefile
+++ linux-5.4/GNUmakefile
@@ -198,11 +198,6 @@
 
 ifeq ($(is-oot-kmod),)
 	@# Skip config diff for out-of-tree module builds
-    ifeq ($(words $(wildcard $(sort $(AVM_KCONFIG_CONFIG) $(KCONFIG_CONFIG)))),2)
-	$(if $(filter-out $(no-config-diff-targets),$@),\
-		@scripts/diffconfig $(AVM_KCONFIG_CONFIG) $(KCONFIG_CONFIG) | \
-		 sed -re '1{s|^|Linux config has changed compared to $(AVM_KCONFIG_CONFIG):\n|}' >&2)
-    endif
 else
 	$(if $(filter modules_install,$@),\
 		@$(MAKE) --no-print-directory modules_install.post)
```
Klar muss man sie bei einigen anderen source files anpassen aber das ist ja normal.


https://github.com/Freetz-NG/freetz-ng/discussions/1063#discussioncomment-11158942
und danach hast du es ja bei der 6670 ja behoben. Und daher laufen die 6591 und 6660 auch durch, weil du es von der 6670 kopiert hast.

**Reply by Conan179** [reply]
@fda77 Sie zahlen nicht genug 🤣

**Reply by SaMMy-lacht** [reply]
Bei der 7530AX fehlen dir ca 11 Zeilen und wie gesagt der 310 fehler! Dan könne man auch diese mit Kernel nutzen

**Reply by SaMMy-lacht** [reply]
Was redest du da überhaupt!

**Reply by SaMMy-lacht** [reply]
Wen die Fehler gefix sind, werde ich meine beweise ja auch freigeben. Sie liegen ja da nicht ohne Grund auf dem Server 
https://www.mediafire.com/folder/wunops45lxl4j/Beweis

**Reply by fda77** [reply]
Also ich blick nicht ganz durch... bei der 7530ax kann man doch gar keine module auswählen. Da ist doch im osp irgend was anders und daher deaktiviert.
Das mit dem "diffconfig" hat avm letzte zeit geändert, für kernel 5.4 un  5.15 identisch sein. Für welche sourcen in freetz soll das kaputt sein?

Oben dein video hat 7590AX im Dateinamen, aber das ist auf jeden fall quatsch, da die 7590AC von der ich ein screenshot gepostet hat die gleichen sourcen nutzt und funktioniert...

**Reply by SaMMy-lacht** [reply]
Weil dir 11 Zeilen fehlen in deinen Patchen! Und der 310 Fehler ansonsten läuft die source 1a von der 7530AX und das schon seit 7.5und 7.8 und auch bei 8.0.

![7530AX3](https://github.com/user-attachments/assets/d4ad4fcc-0ce9-446a-b881-6b200740fec8)
![7530AX2](https://github.com/user-attachments/assets/08abb870-ff15-402e-a63d-e12075015729)
![7530AX1](https://github.com/user-attachments/assets/4718c3d4-5a64-43e4-8070-888c98cb2983)


Und mach, was du willst, mit dem pipefail Fehler. Da du es ja nicht glauben willst, lasse ich es nun. 
Und Unterstellungen muss ich mir auch nicht gefallen lasse!

**Comment by SaMMy-lacht** [comment]
Wenn die Leute also fda77 https://github.com/sponsors/fda77 hier mehr zahlen würden, dann würde er es auch mit den Fehlern lassen. Willst du das damit sagen?

**Reply by Conan179** [reply]
Es ist kein verstecktes abo wir bei "freunden".

**Reply by SaMMy-lacht** [reply]
LoL.

**Comment by fda77** [comment]
Ich hab absolut keine Lust alle kernel durchzutesten! 
Den der 7590ac vorhin lief ja, daher geh ich davon aus dass du dich bei den anderen genau so "geirrt" hast.
7530ax: mach halt einen PR, ich hab das gerät nicht und verplemper damit nicht meine zeit

**Reply by SaMMy-lacht** [reply]
Nein, habe ich nicht! Man bekommt den pipefail Fehler! Aber wie gesagt mach was du willst.

**Reply by fda77** [reply]
Ich mach einfach das was ich am besten kann

**Reply by SaMMy-lacht** [reply]
Da das Betriebssystem Fedora sehr tolerant ist im Bereich Fehlern. Schaut das ganze aber mit Mint, Ubuntu und Debian ein wenig anderes aus. Den dort existiert halt der pipefail Fehler weiterhin. Damit ist freetz-ng in einem weiten Punkt inkompatibel geworden für andere Linux Betriebssysteme.

**Reply by fda77** [reply]
Ich hab dir doch gestern schon klar gemacht dass mich deine sogenannten Fehlermeldungen nicht interessieren!
Beheb den Fehler (PR) oder behalt ihn für dich!! Ich geh eh davon aus dass du den mit deinem Script selbst verursacht hast

**Comment by michael0815000** [comment]
Ich denke, ich habe auch das Problem mit der Option pipefail.
Ich versuche ein Image mit FOS 08.00 für die 7590 auf Ubuntu 20.04.6 mit freetz-ng Commit 865cd859 zu bauen. In derselben Umgebung konnte ich am 24.09.2024 mit Commit 033ccec9d8 noch problemlos ein Image für FOS 07.59 bauen. Ich habe eigentlich nichts außer der FOS-Version an der Konfiguration geändert.

```shell
make -f ./scripts/Makefile.fwinst obj=firmware __fw_modinst
  /usr/bin/bash ./scripts/depmod.sh true 4.9.325 ""
  OBJCOPY adding debug links to stripped kernel modules
set -eu; set -o pipefail; \
    trap 'rm -rf /home/freetz/7590/source/kernel/ref-grx5-7599_08.00/usr/lib/debug/.tmp.stage.' EXIT; \
    ( cd /home/freetz/7590/source/kernel/ref-grx5-7599_08.00/usr/lib/debug/.tmp.stage. && find */ -name '*.ko') | \
    while read module ; do \
        mkdir -p /home/freetz/7590/source/kernel/ref-grx5-7599_08.00/usr/lib/debug/${module%/*}; \
        mv /home/freetz/7590/source/kernel/ref-grx5-7599_08.00/usr/lib/debug/.tmp.stage./${module} /home/freetz/7590/source/kernel/ref-grx5-7599_08.00/usr/lib/debug/${module}; \
        mips-unknown-linux-gnu-objcopy --add-gnu-debuglink=/home/freetz/7590/source/kernel/ref-grx5-7599_08.00/usr/lib/debug/${module} /home/freetz/7590/source/kernel/ref-grx5-7599_08.00/${module}; \
    done
/bin/sh: 1: set: Illegal option -o pipefail
make[3]: *** [GNUmakefile:177: modules_install.post] Fehler 2
make[2]: *** [GNUmakefile:196: modules_install] Fehler 2
make[1]: *** [make/kernel/kernel.mk:246: source/kernel/ref-grx5-7599_08.00/.modules-grx5] Beendet
make: *** [Makefile:47: envira] Beendet
```

**Reply by fda77** [reply]
> noch problemlos ein Image für FOS 07.59 bauen

> source/kernel/ref-grx5-7599_08.00/

Heiland, wieder jemand für den Arschbacken und Kuchen backen das gleiche ist -.- Vielleicht einfach bei 7,59 bleiben wenns eh gleich ist??!

Sieht jedenfalls soso aus als ob das "set" aus den AVM-Sourcen kommt. Was ist denn /bin/sh, also deine Shell

**Reply by michael0815000** [reply]
So, bin nicht bei 7.59 geblieben. Habe zunächst die Arschbacken zusammengekniffen, dann einen Kuchen gebacken, danach das Image fertig gebaut und es gerade erfolgreich auf die 7590 geladen.

Wen's interessiert: Habe in meiner Ubuntu-Build-Umgebung einen Workaround gebastelt. "Shell" war das Stichwort. 

sudo rm /bin/sh
sudo ln -s /bin/bash /bin/sh

Danach konnten die Skripte nicht mehr die Bourne-Shell verwenden, die mit "-o pipefail" auf meinem Ubuntu nicht klarkommt, sondern wurden auf die Bash umgeleitet.

Auf jeden Fall sage ich danke für den großen Entwicklungseinsatz, auch wenn ich das alles sicher nicht in Gänze beurteilen kann.
-.-

**Comment by fda77** [comment]
Die Preisfrage wäre ja ob die letzten Änderungen halfen.
Und was denn deinen Shell ursprünglich war, und ob das so Standard ist oder du es für irgend etwas anderes so geändert hast

**Comment by michael0815000** [comment]
Es hatte nach Deinem Commit b1edb5b noch nicht funktioniert.
Meine Standard-Shell ist die Bash, aber in einigen (oder vielen?) Skripten steht in der ersten Zeile "#!/bin/sh", was bei mir ein Link auf /bin/dash war.
Beispiel: ./make/pkgs/mod/files/root/usr/lib/mww/do_update_handler.sh (vgl. https://github.com/Freetz-NG/freetz-ng/discussions/1012#discussioncomment-10031539)

Ich habe jetzt den Link wieder umgebogen.

freetz@freetz-linux:~/7590$ ll /bin/sh
lrwxrwxrwx 1 root root 9 Nov  7 20:35 /bin/sh -> /bin/dash

habe dann "make kernel-clean" ausgeführt und konnte so den Fehler beim "make" reproduzieren.
```
freetz@freetz-linux:~/7590$ echo $0
-bash
freetz@freetz-linux:~/7590$ set -o pipefail
freetz@freetz-linux:~/7590$ /bin/sh
$ echo $0
/bin/sh
$ set -o pipefail
/bin/sh: 1: set: Illegal option -o pipefail
```

**Reply by fda77** [reply]
Der Befehl ist  in der Datei in der die SHELL nun auch gesetzt wird, das sollte so funkionieren.
Sicher dass der Patch auch angewendet wurde? Im Zweifel: "make kernel-recompile"

Es gibt hier um ein Script aus den AVM sourcen, und avm ist wohl der Meinung dass "bash" bei jeden default ist.

Unter "files/" ist was komplett anderes, das sind Datei auf der Box, und dort läuft die minimale shell der Busybox

Ich hab leider kein Ubuntu mit Compiler etc da

**Reply by fda77** [reply]
Dash ist wohl seit ubuntu 6 default! https://wiki.ubuntu.com/DashAsBinSh
Dort steht auch man soll SHELL in der Makefile setzen

Die Neuerung war also von AVM in den neuen aka "es hat sich nichts geändert" sourcen. Prerequisites gibts von avm nich
